program-runtime/src/invoke_context.rs

```diff
@@ -1012,6 +1015,21 @@ impl<'a> InvokeContext<'a> {
     pub fn get_sysvar_cache(&self) -> &SysvarCache {
         &self.sysvar_cache
     }
+
+    // Push a sibling instruction
+    pub fn add_sibling_instruction(&mut self, instruction: Instruction) {
+        self.sibling_instructions.push(instruction);
+    }
+
+    /// Get a sibling instruction, reverse ordered or last added is index 0
+    pub fn get_sibling_instruction(&self, index: usize) -> Option<&Instruction> {
```

Comment:
This can only access _prior executed_ sibling instructions, right?  Perhaps we can make the function name communicate that?

Reply:
Something like `get_processed_sibling_instructions`?  Any suggestions?

Reply:
`processed` might get confused with commitment somehow :thinking: 

Maybe `prior`, `previous`, `preceding`?  @CriesofCarrots, halp!

Reply:
`get_succeeded_sibling_instructions`?
`get_completed_sibling_instructions`?

Reply:
any extra in the name should reinforce that the instruction was successful and completed

Reply:
I actually don't mind `processed`, what with `process_instruction`. And that is essentially what processed commitment is referring to. However, if you think that's ambiguous, I also like `completed`

Reply:
I suppose `executed` is another option (referring to Justin's refactor here: https://github.com/solana-labs/solana/pull/22215, which refers to transactions, but similar meaning)

Reply:
Hmm, I'd like to keep the `processed` terminology when it comes to instructions, especially due to `process_instruction`.

Reply:
Cool, as I said, I'm fine with `processed`

Reply:
I was hoping for something that more explicitly indicated the positional restriction, I suppose it's implied by processed.  Probably going to be a point of confusion until the community knowledge base gets the hang of it.

Reply:
Another rename, doesn't really address your concern but I think it makes the terminology more consistent with previous work.  Open to suggestions for how to convey the positional restrictions.

Reply:
The rename suggests that I misunderstood the PR description and that this syscall will *not* work for tx-level instructions, only between CPI ix within the same invoking ix. Is that correct?

Reply:
That is correct, just inner instructions within the same top level instruction.

Reply:
Ah, okay. Bit of a bummer to have to use different apis depending on call depth, but can make it work. Thx!